### PR TITLE
feat(ctb): Make `GENESIS_BLOCK_NUMBER` public

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Makes `GENESIS_BLOCK_NUMBER` public, requested by @ajsutton

## Summary by CodeRabbit

- **New Features**
  - Introduced a `GENESIS_BLOCK_NUMBER` for enhanced contract transparency and accessibility.

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #8355** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8355?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8355?utm_source=stack-comment).

**Review by ajsutton — APPROVED** [review]
🎉

**Review by refcell — APPROVED** [review]

File: op-bindings/bindings/outputbisectiongame_more.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ const OutputBisectionGameStorageLayoutJSON = "{\"storage\":[{\"astId\":1000,\"co
...
 
 func init() {
 	if err := json.Unmarshal([]byte(OutputBisectionGameStorageLayoutJSON), OutputBisectionGameStorageLayout); err != nil {
```
[the remark this answers is not in the document]
Comment (clabby) [reply op-bindings/bindings/outputbisectiongame_more.go:19]: Thanks rabbit, these are autogen'd

File: packages/contracts-bedrock/src/dispute/OutputBisectionGame.sol [hunk 2]
```diff
@@ -43,12 +43,12 @@ contract OutputBisectionGame is IOutputBisectionGame, Clone, ISemver {
     /// @notice An onchain VM that performs single instruction steps on a fault proof program trace.
     IBigStepper public immutable VM;
 
+    /// @notice The genesis block number
+    uint256 public immutable GENESIS_BLOCK_NUMBER;
```
Comment (tynes) [inline packages/contracts-bedrock/src/dispute/OutputBisectionGame.sol:47]: This should be internal with a snake get public getter

Comment (ajsutton) [reply packages/contracts-bedrock/src/dispute/OutputBisectionGame.sol:47]: That's inconsistent with the rest of the API (like the `VM` var right above this one) so I suspect we need a wider scale update across both `OutputBisectionGame` and `FaultDisputeGame` which both use public constants. Given this is blocking me and these contracts aren't going to be deployed anytime soon, I've logged https://github.com/ethereum-optimism/client-pod/issues/258 and will merge this.

**Pull request merged**